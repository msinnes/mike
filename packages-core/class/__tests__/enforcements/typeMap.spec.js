const typeLoader = require('../../src/loaders/type');

const typeMapEnforcement = require('../../src/enforcements/typeMap');

describe('typeMapEnforcement', () => {
  it('should be a function', () => {
    expect(typeMapEnforcement).toBeDefined();
    expect(typeMapEnforcement).toBeInstanceOf(Function);
  });

  it('should not throw an error if input not a map of types', () => {
    expect(() => {
      typeMapEnforcement({
        field1: typeLoader('name1', () => {}),
      });
    }).not.toThrow();
  });

  it('should throw an error if an input is not a map of types', () => {
    expect(() => {
      typeMapEnforcement([]);
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      typeMapEnforcement({
        field1: 'string',
        field2: typeLoader('name1', () => {}),
      });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      typeMapEnforcement({
        field1: 'string',
      });
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      typeMapEnforcement();
    }).toThrowErrorMatchingSnapshot();
  });
});
