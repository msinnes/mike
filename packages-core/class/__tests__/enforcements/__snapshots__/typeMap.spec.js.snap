// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeMapEnforcement should throw an error if an input is not a map of types 1`] = `"InternalError: Expected a map of types"`;

exports[`typeMapEnforcement should throw an error if an input is not a map of types 2`] = `"InternalError: Expected a map of types"`;

exports[`typeMapEnforcement should throw an error if an input is not a map of types 3`] = `"InternalError: Expected a map of types"`;

exports[`typeMapEnforcement should throw an error if an input is not a map of types 4`] = `"InternalError: Expected a map of types"`;
